### PR TITLE
Refactor decode-transaction to decode-transaction-message

### PR DESCRIPTION
**Pull request by mcintyre94**
This PR updates the library helper `decodeTransaction` to a new `decodeTransactionMessage`.

Previously this took as input a serialized transaction, and returned the `Transaction` object. It used the `rpc` to look up the address lookup tables and use that to decompile the transaction.

It now takes as input a `TransactionMessage` and returns the `NewTransaction` object. It continues to use the `rpc` to look up the address lookup tables, which are passed to `decompileTransactionMessage`.

This makes it a bit less of an end-to-end helper, but I think isolates the useful functionality better. If you have a serialised transaction you can deserialize it to `NewTransaction` (message bytes + signatures) without knowing anything about lookup tables. If that's all you need, then you don't need this helper. You can also deserialize the message bytes to `CompiledTransactionMessage` without lookup tables. If you need to further convert it to the `TransactionMessage` format, then this helper will fetch the lookup tables for you and decompile it.

### CODE REVIEW

**Comment by mcintyre94** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2468?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2468** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2467** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2467?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2465** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2465?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2464** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2464?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2463** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2463?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2462** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2462?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2459** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2459?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2458** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2458?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2457** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2457?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2456** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2456?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2455** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2455?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2454** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2454?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2453** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2453?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2451** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2451?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2450** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2450?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mcintyre94 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by buffalojoec — COMMENTED** [review]
The tests appear the same, just with minor name changes - like the function name - but is there any way to make these diffs a little better? Maybe two commits in this PR that can be squashed? Not sure if that helps the blame down the road.

File: packages/library/src/decode-transaction-message.ts [hunk 1]
```diff
@@ -41,34 +38,29 @@ async function fetchLookupTables(
...
+    const addressesByLookupTableAddress =
         lookupTableAddresses.length > 0 ? await fetchLookupTables(lookupTableAddresses, rpc, fetchAccountsConfig) : {};
 
-    return getTransactionDecoder({
-        addressesByLookupTableAddress: fetchedLookupTables,
-        lastValidBlockHeight,
-    }).decode(encodedTransaction);
+    return decompileTransactionMessage(compiledTransactionMessage, {
+        addressesByLookupTableAddress,
+        lastValidBlockHeight: lastValidBlockHeight,
```
Comment (lorisleiva) [inline packages/library/src/decode-transaction-message.ts:64]: nit 

```suggestion
        lastValidBlockHeight,
```

**Review by lorisleiva — APPROVED** [review]

**Comment by mcintyre94** [comment]
> The tests appear the same, just with minor name changes - like the function name - but is there any way to make these diffs a little better? Maybe two commits in this PR that can be squashed? Not sure if that helps the blame down the road.

The PR is all 1 commit already unfortunately! I'm not sure if you can change git's threshold of refactor-vs-new, I'll see if I can find anything obvious. Unfortunately the test code is changed quite significantly here despite the tests themselves being very similar, because there's a second layer of decoding that's stripped away and that was mocked in the tests.

**Comment by mcintyre94** [comment]
Hmm can't find anything! It looks like git uses some similarity index, and if it's >50% different then it calls it new. There are config params on `git diff` that you can set: https://git-scm.com/docs/diff-config#Documentation/diff-config.txt-diffrenames. ChatGPT helpfully imagined that you might be able to pass a % in to override the 50% which would be cool! 

But I don't think there's anything you can do to make git or github recognise the commits differently.

**Comment by buffalojoec** [comment]
> Hmm can't find anything! It looks like git uses some similarity index, and if it's >50% different then it calls it new. There are config params on `git diff` that you can set when you view a diff locally: https://git-scm.com/docs/diff-config#Documentation/diff-config.txt-diffrenames. ChatGPT helpfully imagined that you might be able to pass a % in to override the 50% which would be cool!
> 
> But I don't think there's anything you can do to make git or github recognise the commits differently.
> 
> Using separate commits might help, but it'd have to be different PRs because we squash on merge and Git 'optimises' things. [See last time I went down that rabbit hole!](https://github.com/solana-labs/solana-web3.js/pull/2423#issuecomment-2036905509)

Oh well! Appreciate the detailed search though!

screw it

**Review by buffalojoec — APPROVED** [review]

**Comment by mcintyre94** [comment]
### Merge activity

* **Apr 10, 9:58 AM EDT**: @mcintyre94 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2468).

**Pull request merged**